### ISSUE TITLE
spec-char-escape's regex check against multiple spaces

### ISSUE THREAD

**Issue by jonknapp** [htmllint/htmllint#118]
What is the multiple space regex test for in the `spec-char-escape` rule? My original thought was that it was checking to make sure multiple spaces are not found in a row in the html file, though I'm not sure if it's something else or not.

I recently turned on the `spec-char-escape` rule in a project and I've been flooded with failures due to the double spacing error: `line 2	col 1	text contains improperly escaped characters: multiple adjacent spaces (E023)`

**Comment by jonknapp** [comment]
Just checked the test suite and I guess it does refer to multiple text characters in a row.

https://github.com/htmllint/htmllint/blob/7ef9ff922f863f7cbe253fadb6492826ad175593/test/functional/spec-char-escape.js

Would it be a bad idea to move this to a separate rule? I like the idea of finding unescaped special characters, but think it's a separate need than removing unneeded whitespace inside elements.

**Comment by KrekkieD** [comment]
This is an issue for me, it invalidates the following HTML due to the spacing between the end of the `html` tag and the start of the `head` tag:

```html
<!DOCTYPE html>
<html lang="en">
    <head>
        <title>Dingles</title>
    </head>
    <body></body>
</html>
```

I'm not sure how the spacing regex will improve code. Perhaps it should exclude leading and trailing whitespace?

**Comment by KrekkieD** [comment]
The test case tests on an empty element (empty `p` with nothing but spaces). I think that should be a rule on its own, or at least not part of the spec-char-escape rule as double spaces has nothing to do with special characters.

**Comment by mlochbaum** [comment]
I agree that the multiple-space test has nothing to do with special characters. It's strange for other reasons, too, since it will fail text which is broken into lines and indented in the source, or even text with two spaces after a period. I'm just going to remove it now, but will consider whether there should be a test for empty paragraphs or something like that.
